storage/registers.go

```diff
@@ -0,0 +1,35 @@
+package storage
+
+import (
+	"io"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// Todo: Update interface after https://github.com/onflow/flow-go/pull/4656/ gets merged
+
+// Registers defines methods for the register index.
+type Registers interface {
+	RegisterReader
+	RegisterWriter
+	io.Closer
+}
+
+// RegisterReader defines read-only operations on the register index.
+type RegisterReader interface {
+	// Get register by the register ID at a given block height.
+	//
+	// If the register at the given height was not indexed, returns the highest
+	// height the register was indexed at.
+	// An error is returned if the register was not indexed at all. Expected errors:
+	// - storage.ErrNotFound if the register was not found in the db
```

Comment:
We need to handle the additional two cases described in `storage.ErrOutOfRange`
```suggestion
	// - storage.ErrNotFound if the register was not found in the db
	// - storage.ErrOutOfRange if the queried height is below the root block height or above the last indexed height.
```